### PR TITLE
Drop dependency on httpoison, use hackney directly

### PR DESCRIPTION
**Pull request by michalmuskala**
This removes one dependency from the project, and I don't think it complicates it in any significant way.

### CODE REVIEW

**Comment by paulcsmith** [comment]
I looked at this quickly on my phone and it seems good to me! I'll have to take a closer look on Friday when I have a bit more time but I think it'll be good to merge as-is. Thanks!

File: lib/bamboo/adapters/mailgun_adapter.ex [hunk 1]
```diff
@@ -123,7 +130,7 @@ defmodule Bamboo.MailgunAdapter do
   defp put_text_body(body, %Email{text_body: text_body}), do: Map.put(body, :text, text_body)
 
   defp full_uri(config) do
-    (Application.get_env(:bamboo, :mailgun_base_uri) || @base_uri)
-    <> config.domain <> "/messages"
+    [Application.get_env(:bamboo, :mailgun_base_uri) || @base_uri,
```
Comment (paulcsmith) [inline lib/bamboo/adapters/mailgun_adapter.ex]: It seems strange that this return a list and not a string. Is that just a Hackney thing?

**Comment by paulcsmith** [comment]
Just one question about using a list as the URI. Other than that, this looks great!

File: lib/bamboo/adapters/mailgun_adapter.ex (hunk 1 above)
Comment (michalmuskala) [inline lib/bamboo/adapters/mailgun_adapter.ex]: hackney acceps iolists as url, I can revert it to append.

Comment (paulcsmith) [inline lib/bamboo/adapters/mailgun_adapter.ex]: I think I'd prefer that if it's cool with you :)

**Comment by michalmuskala** [comment]
@paulcsmith updated

**Comment by paulcsmith** [comment]
Thank you! :D Yay for one less dependency

**Pull request merged**